corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -282,6 +305,91 @@ hqDefine("geospatial/js/geospatial_map", [
             }
         };
 
+        function resetMarkersOpacity() {
+            let mapInstance = map.getMapboxInstance();
+            let markers = [];
+            Object.keys(caseGroupsIndex).forEach(itemCoordinates => {
+                const mapMarkerItem = caseGroupsIndex[itemCoordinates];
+                markers.push(mapMarkerItem.item);
+
+                const lineId = getLineFeatureId(mapMarkerItem.item.itemId);
+                if (mapInstance.getLayer(lineId)) {
+                    mapInstance.setPaintProperty(lineId, 'line-opacity', 1);
+                }
+            });
+            // It's necessary to delay obscuring the markers since mapbox does not play nice
```

Comment:
Do you perhaps know why? Is it related to the amount of markers on the map?

Reply:
Not 100% sure, but I think it might be some sort of race condition happening where the second update seems to override the first one.